src/pages/ReimbursementAccount/ReimbursementAccountPage.js

```diff
@@ -214,129 +208,151 @@ class ReimbursementAccountPage extends React.Component {
-    }
-
-    /**
-     * @param {String} fieldName
-     * @param {*} defaultValue
-     *
-     * @returns {*}
-     */
-    getDefaultStateForField(fieldName, defaultValue = '') {
-        return lodashGet(this.props.reimbursementAccount, ['achData', fieldName], defaultValue);
-    }
-
-    /**
-     * We can pass stepToOpen in the URL to force which step to show.
-     * Mainly needed when user finished the flow in verifying state, and Ops ask them to modify some fields from a specific step.
-     * @returns {String}
-     */
-    getStepToOpenFromRouteParams() {
-        switch (lodashGet(this.props.route, ['params', 'stepToOpen'], '')) {
-            case 'new':
-                return CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT;
-            case 'company':
-                return CONST.BANK_ACCOUNT.STEP.COMPANY;
-            case 'personal-information':
-                return CONST.BANK_ACCOUNT.STEP.REQUESTOR;
-            case 'contract':
-                return CONST.BANK_ACCOUNT.STEP.ACH_CONTRACT;
-            case 'validate':
-                return CONST.BANK_ACCOUNT.STEP.VALIDATION;
-            case 'enable':
-                return CONST.BANK_ACCOUNT.STEP.ENABLE;
-            default:
-                return '';
-        }
-    }
-
-    /**
-     * @param {String} currentStep
-     * @returns {String}
-     */
-    getRouteForCurrentStep(currentStep) {
-        switch (currentStep) {
-            case CONST.BANK_ACCOUNT.STEP.COMPANY:
-                return 'company';
-            case CONST.BANK_ACCOUNT.STEP.REQUESTOR:
-                return 'personal-information';
-            case CONST.BANK_ACCOUNT.STEP.ACH_CONTRACT:
-                return 'contract';
-            case CONST.BANK_ACCOUNT.STEP.VALIDATION:
-                return 'validate';
-            case CONST.BANK_ACCOUNT.STEP.ENABLE:
-                return 'enable';
-            case CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT:
-            default:
-                return 'new';
-        }
+        return achData.state === BankAccount.STATE.PENDING || _.contains([CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT, ''], getStepToOpenFromRouteParams(route));
     }
 
-    /**
-     * Returns true if a VBBA exists in any state other than OPEN or LOCKED
-     * @returns {Boolean}
-     */
-    hasInProgressVBBA() {
-        const achData = lodashGet(this.props.reimbursementAccount, 'achData', {});
-        return achData.bankAccountID && achData.state !== BankAccount.STATE.OPEN && achData.state !== BankAccount.STATE.LOCKED;
-    }
+    // The first time we open this page, props.reimbursementAccount is either not available in Onyx
+    // or only partial data loaded where props.reimbursementAccount.achData.currentStep is not available
+    // Calculating shouldShowContinueSetupButton on first page open doesn't make sense, and we should recalculate
+    // it once we get the response from the server the first time in componentDidUpdate.
```

Comment:
This doc is out of date now since it references `componentDidUpdate` which is no longer used.